test/test_torch.py

```diff
@@ -7595,6 +7618,17 @@ def test_masked_fill(self):
         dst2.masked_fill_(dst2 > 0, val)
         self.assertEqual(dst, dst2, 0)
 
+        # test bool tensor case
+        for device in torch.testing.get_all_device_types():
+            x = torch.tensor([True, True, False, False], dtype=torch.bool, device=device)
```

Comment:
can you check that there is a warning with byte tensors?  Presumably you also want to guard any existing calls in tests that may exist so we don't spew warnings.